mlte/property/property.py

```diff
@@ -35,91 +36,60 @@ def __init__(self, name: str, description: str, rationale: str):
+    def to_model(self) -> PropertyModel:
         """
-        Save a Property instance to a JSON document.
+        Return a Property as a model.
 
-        :return: The document
-        :rtype: Dict[str, str]
+        :return: The property as its model.
+        :rtype: PropertyModel
         """
-        return {
-            "name": self.name,
-            "repr": repr(self),
-            "description": self.description,
-            "rationale": self.rationale,
-        }
-
-    @staticmethod
-    def _from_json(document: Dict[str, str]) -> Property:
+        return PropertyModel(
+            name=self.name,
+            description=self.description,
+            rationale=self.rationale,
+        )
+
+    @classmethod
+    def from_model(cls, model: PropertyModel) -> Property:
         """
-        Load a Property instance from a JSON document.
+        Load a Property instance from a model.
 
-        :param document: The document for the saved property
-        :type document: Dict[str, str]
+        :param model: The model with the Property info.
+        :type model: PropertyModel
 
         :return: The loaded property
         :rtype: Property
         """
-        return _load_from_document(document)
-
-    def __repr__(self) -> str:
-        """Return the representation needed to reconstruct the object."""
-        return f"{self.name}()"
-
-
-def _get_class_name(property_repr: str) -> str:
-    """
-    Extract the class name from the property representation.
-
-    :param property_repr: The representation
-    :type property_repr: str
-
-    :return: The class name
-    :rtype: str
-    """
-    return property_repr[: property_repr.index("(")]
-
-
-def _load_from_document(document: Dict[str, str]) -> Property:
-    """
-    Load a Property instance from its identifier.
-
-    :param name: The name of the property
-    :type name: str
-
-    :return: The loaded property
-    :rtype: Property
-    """
-    if "name" not in document or "repr" not in document:
-        raise RuntimeError("Saved property is malformed.")
-    property_repr = document["repr"]
-    rationale = document["rationale"]
-
-    # Extract the classname from the call
-    classname = _get_class_name(property_repr)
-
-    # Load the class type from the module
-    properties_package_name = "mlte.property"
-    properties_module = importlib.import_module(
-        properties_package_name, package="mlte"
-    )
-    for submodule_info in pkgutil.iter_modules(
-        properties_module.__path__, properties_module.__name__ + "."
-    ):
-        submodule = importlib.import_module(
-            submodule_info.name, package=properties_package_name
+        if model.name == "":
+            raise RuntimeError(
+                "Property is malformed, it does not have a valid name."
+            )
+        classname = model.name
+
+        # Load the class type from the module
+        properties_package_name = "mlte.property"
```

Comment:
This is likely similar to the machinery that we might add to `Value.from_model()` to dynamically infer the value type.

Reply:
Correct, this could be a good starting point. The main complication with Values is that extended values can be in any package, so the handling of the full path will be needed, and the code will be a bit trickier.